### PR TITLE
File movings

### PR DESCRIPTION
**Pull request by ykate1998**
# Description
 
Adding the CNN Image Retrieval model implementation based on the following papers:

[1] Fine-tuning CNN Image Retrieval with No Human Annotation,
Radenović F., Tolias G., Chum O., TPAMI 2018 [arXiv]
https://arxiv.org/abs/1711.02512

[2] CNN Image Retrieval Learns from BoW: Unsupervised Fine-Tuning with Hard
Examples, Radenović F., Tolias G., Chum O., ECCV 2016 [arXiv]
https://arxiv.org/abs/1604.02426

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] A new research paper code implementation

## Tests

Tests for the added modules are provided.

**Test Configuration**:

Same requirements as for the DELF library.

### CODE REVIEW

File: research/delf/delf/__init__.py [hunk 1]
```diff
@@ -35,5 +35,4 @@
 from delf.python import detect_to_retrieve
 from delf.python import training
 from delf.python.training import model
-from delf.python.training import datasets
```
Comment (andrefaraujo) [inline research/delf/delf/__init__.py:38]: just wanted to double-check: after doing this change, the code continues to work, right? I thought some import like this in the top init could still be needed.

File: research/delf/delf/python/pooling_layers/pooling.py [hunk 2]
```diff
@@ -19,19 +19,16 @@
 
 class MAC(tf.keras.layers.Layer):
   """Global max pooling (MAC) layer.
-
```
Comment (andrefaraujo) [inline research/delf/delf/python/pooling_layers/pooling.py:22]: there are many blank-line removals in this file, can you please revert those?

**Review by andrefaraujo — CHANGES_REQUESTED** [review]
Thanks Kate! Left some comments. Also, one question: for this PR, would it be possible/easy to remove the files "global_features_utils.py", "whiten.py" and "whiten_test.py"? They are unrelated to file movings, so this PR would be cleaner / simpler without them.

File: research/delf/delf/python/pooling_layers/pooling.py (hunk 2 above)
Comment (ykate1998) [reply research/delf/delf/python/pooling_layers/pooling.py:22]: Thank you for pointing this out, must have missed some code formatting issue. Should be fixed now

File: research/delf/delf/__init__.py (hunk 1 above)
Comment (ykate1998) [reply research/delf/delf/__init__.py:38]: The code was running, but I now added a new updated import just to make sure it works the same way

**Comment by ykate1998** [comment]
Deleted the unnecessary modules you mentioned. Thank you :)

**Review by andrefaraujo — APPROVED** [review]
Thanks!

**Pull request merged**